File: Algorithms/Binary Search/binarySearch.py
def binary_search(arr, key): 
    low = 0
    high = len(arr) - 1
    mid = 0
  
    while low <= high: 
        mid = (high + low)

        if arr[mid] < key: 
            low = mid + 1
  
        # If key is greater than the middle element, ignore left half 
        elif arr[mid] > key: 
            high = mid - 1
  
        # If key is smaller than the middle element, ignore right half
        else: 
            return mid 
  
    # when the element is not present 
    return -1
  
  
arr = [ 5, 7, 9, 10, 13 ] 

key = 10
result = binary_search(arr, key) #stores the index of the key, -1 if not found
print(result)

key = 11
result = binary_search(arr, key) #stores the index of the key, -1 if not found
print(result)

File: Algorithms/Binary Search/binarySearch.java
package college;

public class binarySearch {

	public static void main(String[] args) {
		int ar[] = {1,2,5,8,12,27, 31,42};
		int element = 27;
		int ans = binarySearch(ar, 0, 7, element);
		if (ans!=-1)
				System.out.println("element found at "+ans);
		else
			System.out.println("not found");
	}
	
	static int binarySearch(int[] ar, int start, int end, int element) {
		if (start>end)
			return -1;
		int mid = (start + end)/2;
		if (element == ar[mid])
			return mid;
		else if(element > ar[mid]) 
			 return binarySearch(ar, mid+1, end, element);
		else if(element < ar[mid])
			return binarySearch(ar, start, mid-1, element);
		return mid;
		
	}

}
